xds/src/main/java/io/grpc/xds/ClusterImplLoadBalancer.java

```diff
@@ -385,6 +385,7 @@ private RequestLimitingSubchannelPicker(SubchannelPicker delegate,
       public PickResult pickSubchannel(PickSubchannelArgs args) {
         args.getCallOptions().getOption(ClusterImplLoadBalancerProvider.FILTER_METADATA_CONSUMER)
             .accept(filterMetadata);
+        args.getPickDetailsConsumer().addOptionalLabel("grpc.xds.cluster", cluster);
```

Comment:
Shouldn't this be flag protected?

Reply:
Otel ignores this and the gRFC [has no env variable protection](https://github.com/grpc/proposal/blob/296f5ec3eaad06eee7f0b37227890f8e492e7311/A89-xds-cluster-metric-label.md#temporary-environment-variable-protection) because you use an API to enable the functionality.